plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConverter.kt

```diff
@@ -5,23 +5,37 @@
 
 package org.jetbrains.kotlin.formver.conversion
 
+import org.jetbrains.kotlin.fir.expressions.FirExpression
 import org.jetbrains.kotlin.fir.expressions.FirStatement
 import org.jetbrains.kotlin.formver.embeddings.TypeEmbedding
 import org.jetbrains.kotlin.formver.embeddings.VariableEmbedding
+import org.jetbrains.kotlin.formver.viper.MangledName
 import org.jetbrains.kotlin.formver.viper.ast.Exp
 import org.jetbrains.kotlin.formver.viper.ast.Stmt
 import org.jetbrains.kotlin.formver.viper.domains.UnitDomain
 
 abstract class BaseStmtConverter(
     private val methodCtx: MethodConversionContext,
     private val seqnCtx: SeqnBuildContext,
+    override val nameResolver: NameResolutionContext = SimpleNameResolver()
 ) : StmtConversionContext, SeqnBuildContext by seqnCtx, MethodConversionContext by methodCtx {
     override fun convert(stmt: FirStatement): Exp = stmt.accept(StmtConversionVisitor, this)
-    override fun newBlock(): StmtConverter = StmtConverter(this, SeqnBuilder())
+    override fun newBlock(): StmtConverter = StmtConverter(this, SeqnBuilder(), nameResolver)
     override fun withResult(type: TypeEmbedding): StmtWithAnonResultConverter {
         val newResultVar = newAnonVar(type)
         addDeclaration(newResultVar.toLocalVarDecl())
-        return StmtWithAnonResultConverter(this, seqnCtx, newResultVar)
+        return StmtWithAnonResultConverter(this, seqnCtx, newResultVar, nameResolver)
+    }
+
+    override fun convertAndStore(exp: FirExpression): Exp.LocalVar {
+        val convertedExp = convert(exp)
+        return if (convertedExp is Exp.LocalVar) {
+            convertedExp
+        } else {
+            val ctx = withResult(embedType(exp))
+            ctx.captureResult(convertedExp)
+            ctx.resultVar.toLocalVar()
```

Comment:
Would `withResult(embedType(exp)) { captureResult(convertedExp) }` work?

Actually, perhaps even `return (converted as? Exp.LocalVar) :? withResult(..) {..}`?

Reply:
yes, can I make `withResult` returning a `LocalVar` instead of an Exp? Otherwise I need to do a cast

Reply:
Yeah, go for it, that would make more sense anyway